vector/src/main/java/im/vector/app/features/home/room/detail/TimelineFragment.kt

```diff
@@ -1922,30 +1925,36 @@ class TimelineFragment @Inject constructor(
                         }
                     })
             if (!isManaged) {
-                if (title.isValidUrl() && url.isValidUrl() && URL(title).host != URL(url).host) {
-                    MaterialAlertDialogBuilder(requireActivity(), R.style.ThemeOverlay_Vector_MaterialAlertDialog_NegativeDestructive)
-                            .setTitle(R.string.external_link_confirmation_title)
-                            .setMessage(
-                                    getString(R.string.external_link_confirmation_message, title, url)
-                                            .toSpannable()
-                                            .colorizeMatchingText(url, colorProvider.getColorFromAttribute(R.attr.vctr_content_tertiary))
-                                            .colorizeMatchingText(title, colorProvider.getColorFromAttribute(R.attr.vctr_content_tertiary))
-                            )
-                            .setPositiveButton(R.string._continue) { _, _ ->
-                                openUrlInExternalBrowser(requireContext(), url)
-                            }
-                            .setNegativeButton(R.string.action_cancel, null)
-                            .show()
-                } else {
-                    // Open in external browser, in a new Tab
-                    openUrlInExternalBrowser(requireContext(), url)
+                when {
+                    url.containsRtLOverride() || (title.isValidUrl() && url.isValidUrl() && URL(title).host != URL(url).host) -> {
+                        displayUrlConfirmationDialog(title.ensureEndsLeftToRight(), url.filterDirectionOverrides())
```

Comment:
we have to ensure the text returns to LtR at the end with `title.ensureEndsLeftToRight()`, otherwise the override character will cause the rest of the string resource to also become inverted 

| WITHOUT LTR OVERRIDE |
| --- |
|![Screenshot_20220526_124304](https://user-images.githubusercontent.com/1848238/170481448-84bcb53d-666b-4188-b09f-4f29f745e90c.png)

Reply:
`url.filterDirectionOverrides()` will also be used to open the url, if the user accepts. Will it work fine? Or considerer adding `url` as a new param.

Reply:
ah! no it should be a separate parameter, will update :+1:

Reply:
67f1929784a0554fdd90ce099d160fd4b9a6d98a